devlab/tests/generate_load.py

```diff
@@ -437,13 +437,15 @@ def emulate_vm_states(self):
 
     def generate_vm_state_list(self):
         data = {}
-        for vm in self.novaclient.servers.list():
-            vm_name = vm.name
-            index = self.novaclient.servers.list().index(vm)
-            while self.novaclient.servers.list()[index].status == u'RESIZE':
-                time.sleep(2)
-            vm_state = self.novaclient.servers.list()[index].status
-            data[vm_name] = vm_state
+        for vm in self.novaclient.servers.list(search_opts={'all_tenants': 1}):
+            for i in range(600):
```

Comment:
Magic numbers are considered bad practice in general. Please refactor to something readable.